src/proto/udppacket.cpp

```diff
@@ -21,17 +21,22 @@
 using namespace std;
 using namespace boost::asio::ip;
 
-int UDPPacket::parse(const string &data) {
-    int address_len = address.parse(data);
-    if (address_len == -1 || data.length() < (unsigned int)address_len + 2) {
-        return -1;
+bool UDPPacket::parse(const string &data, size_t &udp_packet_len) {
+    if (data.length() <= 0) {
+        return false;
+    }
```

Comment:
I don't think this is necessary, since it'll fail the `address.parse` call.

Reply:
Just to make it clear.

Reply:
Sure.